File: golang/transposematrix867/transpose-matrix-867.go
package main

import "fmt"

func main() {
	fmt.Println("salam")
}

func transpose(A [][]int) [][]int {
	res := make([][]int, 0)

	for j := 0; j < len(A[0]); j++ {
		col := make([]int, 0)
		for i := 0; i < len(A); i++ {
			col = append(col, A[i][j])
		}
		res = append(res, col)
	}
	return res
}

File: nodejs/transpose-matrix-867.js
'use strict';
/**
 * @param {number[][]} A
 * @return {number[][]}
 */
var transpose = function(A) {
    let res = [];
    for (let i = 0; i < A[0].length; i++) {
        let col = [];
        for (let j = 0; j < A.length; j++) {
            col.push(A[j][i]);
        }
        res.push(col);
    }
    return res;
};

console.log(transpose([[1,2,3],[4,5,6],[7,8,9]]));
console.log(transpose([[1,2,3],[4,5,6]]));

